clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetchCollectorTest.java

```diff
@@ -620,10 +646,49 @@ public void testCollectFetchInitializationOffsetOutOfRangeErrorWithOffsetReset()
         final Fetch<String, String> fetch = fetchCollector.collectFetch(fetchBuffer);
 
         assertTrue(fetch.isEmpty());
+        assertEquals(0, fetch.nextOffsets().size());
         verify(subscriptions).requestOffsetResetIfPartitionAssigned(topicPartition0);
         verify(fetchBuffer).setNextInLineFetch(null);
     }
 
+    @Test
+    public void testReadCommittedWithAbortedTransaction() {
+        buildDependencies(IsolationLevel.READ_COMMITTED);
+        int recordCount = 20;
+        assignAndSeek(topicAPartition0);
+
+
+        // Initial CompletedFetch
+        CompletedFetch initialCompletedFetch = completedFetchBuilder
+            .recordCount(recordCount)
+            .build();
+        fetchBuffer.add(initialCompletedFetch);
+        Fetch<String, String> fetch = fetchCollector.collectFetch(fetchBuffer);
+
+        assertFalse(fetch.isEmpty());
+        assertEquals(recordCount, fetch.numRecords());
+        assertEquals(1, fetch.nextOffsets().size());
+        assertEquals(new OffsetAndMetadata(fetch.numRecords(), Optional.empty(), ""), fetch.nextOffsets().get(topicAPartition0));
+
+
+        // Subsequent CompletedFetch with abort marker
+        Records rawRecords = createTranscactionalRecords(ControlRecordType.ABORT, recordCount);
+        FetchResponseData.PartitionData partitionData = new FetchResponseData.PartitionData()
+            .setRecords(rawRecords);
```

Comment:
I think for the `FetchResponseData` to be correctly formed, we alos have to `setAbortedTransactions`. See, for example, `CompletedFetchTest`.